airootfs/usr/share/vx-img/flash-image.sh

```diff
@@ -78,31 +81,81 @@ for f in "$_path"/*; do
     fi
 done
 
+if [[ -z $_toflash ]]; then
+    echo "Found no image to flash. Exiting..."
+    exit
+fi
+
 echo "Extracting and flashing $_toflash"
 
-echo "What is the expected final size of the image? [64g]:"
+clear
+echo "What is the expected final size of the image, in GB? [64]:"
 read -r _finalsize
 
 if [[ -z $_finalsize ]]; then
-    _finalsize="64g"
+    _finalsize="64"
 fi
 
-$_compression -c -d $_path/"$_filename" | pv -s $_finalsize > /dev/nvme0n1
+_disk="/dev/nvme0n1"
+
+# Get our image size in raw bytes
+_size=$(numfmt --from=iec "${_finalsize}G")
+
+clear
+echo "Found the following disks to flash:"
+while true; do
+
+    # Get a list of all available disks large enough to take our image, sorted by size
+    IFS=$'\n' read -r -d '' -a disks <<< "$(lsblk -x SIZE -nblo NAME,SIZE,TYPE | grep "disk" | awk -v var="$_size" '$2 > var {print $1}')"
+
+    if [[ -z ${disks[0]} ]]; then
+        echo "There are no disks big enough for this image! Exiting..."
+        exit
```

Comment:
what happens when we exit? Will the machine power off, will the messages stay on the screen? If the messages don't stay on the screen might be worth adding a sleep so that the user sees the error

Reply:
oh this is a good point. Right now it will drop the user into a shell, which probably isn't the best user experience :sweat_smile:. I could do something else like jump to a prompt to shutdown, like `There is nothing to do, shutdown? [Y/n]`

Reply:
and if they hit no, drop them into a shell?